src/main/java/com/jcraft/jsch/KeyExchange.java

```diff
@@ -69,6 +69,11 @@
 
   public abstract void init(Session session, 
                             byte[] V_S, byte[] V_C, byte[] I_S, byte[] I_C) throws Exception;
+  public void doInit(Session session, 
```

Comment:
I don't think we should make this public. I'd change it to package private.

Reply:
I've changed it. I've copied init's method signature to make sure that the parameters are identical.